django/db/backends/sqlite3/creation.py

```diff
@@ -12,17 +12,17 @@ class DatabaseCreation(BaseDatabaseCreation):
     def _get_test_db_name(self):
         test_database_name = self.connection.settings_dict['TEST']['NAME']
         can_share_in_memory_db = self.connection.features.can_share_in_memory_db
-        if test_database_name and test_database_name != ':memory:':
-            if 'mode=memory' in test_database_name and not can_share_in_memory_db:
-                raise ImproperlyConfigured(
-                    "Using a shared memory database with `mode=memory` in the "
-                    "database name is not supported in your environment, "
-                    "use `:memory:` instead."
-                )
-            return test_database_name
-        if can_share_in_memory_db:
+        if not test_database_name:
+            test_database_name = ':memory:'
+        if not can_share_in_memory_db and 'mode=memory' in test_database_name:
```

Comment:
Do you think this would be an improvement:
```python
if can_share_in_memory_db:
    if test_database_name == ':memory:'
        return ...
elif 'mode=memory' in test_database_name
    raise ...
```